coding/string_utf8_multilang.hpp

```diff
@@ -129,6 +137,30 @@ class StringUtf8Multilang
     }
   }
 
+  // Used for ordered languages, if you want to do something with priority of that order.
+  template <typename Fn>
+  bool ForEachLanguage(std::vector<std::string> const & languages, Fn && fn) const
+  {
+    auto const & translationPositions = GenerateTranslationPositions();
+
+    base::ControlFlowWrapper<Fn> wrapper(std::forward<Fn>(fn));
+    for (std::string const & language : languages)
+    {
+      int8_t const languageCode = GetLangIndex(language);
+      if (GetLangByCode(languageCode) != kReservedLang)
+      {
+        auto const & translationPositionsIt = translationPositions.find(languageCode);
+        if (translationPositionsIt != translationPositions.end() &&
+            wrapper(languageCode, GetTranslation(translationPositionsIt->second)) ==
+                base::ControlFlow::Break)
+        {
+          return true;
```

Comment:
Why would you return true on break?
Break carries the semantics of stopping, not that of succeeding.

The success flag can be stored if the wrapper if needed.

s/bool/void/, please

Reply:
why cant I return in my signature, was cycle finished by break or not?

Reply:
see for .. else in Python

Reply:
You can't because you have not documented it.
If the question is why you shouldn't then that's because this particular semantics is hard to guess and seems to be tailored for your particular use case. Note that you do not check the return value in tests.

Reply:
ForEachLanguage semantics documented and return value testes